include/zephyr/net/prometheus/pm_collector.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2024 Mustafa Abdullah Kus, Sparse Technology
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ *
+ * @brief prometheus collector APIs.
+ */
+
+#ifndef ZEPHYR_INCLUDE_PM_COLLECTOR_H_
+#define ZEPHYR_INCLUDE_PM_COLLECTOR_H_
+
+#include <zephyr/net/prometheus/pm_metric.h>
+
+#include <stddef.h>
+
+typedef struct {
+	pm_metric_t *metric;
+	struct pm_collector_node *next;
+} pm_collector_node_t;
+
+typedef struct {
+	pm_collector_node_t *head;
+	size_t size;
+} pm_collector_t;
+
+/**
+ * @brief Create a Prometheus collector
+ *
+ * Creates a Prometheus collector to collect metrics.
+ *
+ * @return Pointer to the created collector.
+ */
+pm_collector_t *pm_collector_create(void);
```

Comment:
About the `pm_` prefix, I think we could use `prometheus_` as suggested by @bjarki-trackunit. It is a bit longish but we avoid confusion with power mgmt APIs.